joint_state_publisher/scripts/joint_state_publisher

```diff
@@ -32,6 +32,12 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+import signal
+import sys
+import threading
+
+from python_qt_binding.QtWidgets import QApplication
```

Comment:
This import pulls back in a pyqt/pyside dependency. Could we move the import to within the `try/except`?

Alternatively, could refactor a bit to add helper functions to the `joint_state_publisher_gui` module to construct the `QApplication` singleton and to call `sys.exit(QApplication.instance()._exec())`, to pull the import out of `jsp` entirely, but that might be adding unnecessary complexity for a workaround that will be removed in Noetic anyways.

Reply:
On second thought, it might be worth pulling all four of these imports into that `if use_gui:` or `try:` block, so that there aren't unused imports slowing things down when using the `jsp` in intended headless mode. Seems unlikely to make an appreciable difference, but I thought I'd flag it for your consideration.

Reply:
Absolutely, that's a good call.  I'll move them all, it will make it easier to remove the whole thing for Noetic anyway.

Reply:
Done in f1ed8c9